airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py

```diff
@@ -0,0 +1,116 @@
+
+        url_base = self.format_url_base(host)
+
+        authenticator = SnowflakeJwtAuthenticator.from_config(config)
+
+        check_connection_stream = CheckConnectionStream(url_base=url_base, config=config, authenticator=authenticator)
+        try:
+            records = check_connection_stream.read_records(sync_mode=SyncMode.full_refresh)
+            next(records)
+        except StopIteration:
+            error_message = "There is no stream available for the connection specification provided"
+            raise StopIteration(error_message)
+        except requests.exceptions.HTTPError as error:
+            error_message = error.__str__()
+            error_code = error.args[0]
+            if error_code == 412:
+                error_message = ("SQL execution error for check.\n"
+                                 "The origin of the error is very likely to be:\n"
+                                 "- The configuration provided does not have enough permissions to access the requested database/schema.\n"
+                                 "- The configuration is not consistent (example: schema not present is database.")
+            raise requests.exceptions.HTTPError(error_message)
+
+        return True, None
+
+    @classmethod
+    def format_url_base(cls, host: str) -> str:
+        url_base = host
+        if not host.startswith(cls.HTTP_PREFIX):
+            url_base = f"{cls.HTTP_PREFIX}{host}"
+        return url_base
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        """
+        :param config: A Mapping of the user input configuration as defined in the connector spec.
+        """
+        host = config['host']
+        url_base = self.format_url_base(host)
+        authenticator = SnowflakeJwtAuthenticator.from_config(config)
+
+        table_catalog_stream = TableCatalogStream(url_base=url_base,
+                                                  config=config,
+                                                  authenticator=authenticator)
+        standard_streams = {}
+        for table_object in table_catalog_stream.read_records(sync_mode=SyncMode.full_refresh):
+            standard_stream = TableStream(url_base=url_base,
+                                          config=config,
+                                          authenticator=authenticator,
+                                          table_object=table_object)
+            standard_streams[standard_stream.name] = standard_stream
+
+        push_down_filters_streams = []
+        if 'streams' in config and config['streams']:
+            for stream in config['streams']:
+                parent_stream_name = stream['parent_stream']
+                if parent_stream_name not in standard_streams:
+                    continue
+                parent_stream = standard_streams[parent_stream_name]
+                parent_namespace = stream['parent_namespace']
```

Comment:
Based on the spec, `parent_namespace` is not required however, this code would break if the field is not provided

Reply:
It is concerning push down filters only. The user may specify the namespace. I just updated the parent namespace accordingly Am I misunderstanding something ?